src/languages/typescript.js

```diff
@@ -222,6 +223,14 @@ export default function(hljs) {
       {
         begin: '\\.' + hljs.IDENT_RE, relevance: 0 // hack: prevents detection of keywords after dots
       },
+      {
+        begin: '\\b' + IDENT_RE + ':', relevance: 0 // prevents detection of keywords in property name
+      },
+      {
+        className: 'function',
+        begin: IDENT_RE + '\\s*\\(', returnBegin: true,
+        relevance: 0
+      },
```

Comment:
I'm not sure this is the correct solution here. Many of our built-ins ARE function names (all BUILT_IN_GLOBALS for example)... what specific issues are you having that caused you to add this rule?

Reply:
Better. Sorry for confusion.

Reply:
It turned out that same problem affects methods name too, so for ex.
`delete(): void`
will have `delete` keyword highlighted even when it shouldn't

Reply:
This may be starting to get a little bigger than you were hoping for, but here is what I see off the top of my head:

- Lose the `className`, traditionally we don't highlight function invocation.  I'm open-minded to changing that, but this PR isn't the place to start that journey.
- `returnBegin` looks wrong and probably only works because you have `ARGUMENTS` later to save you from an infinite loop.  Or it could also be our infinite loop detection logic saving the day... there is no reason to return to the beginning here. (unless I'm missing something)
- You probably want to remove that and instead use a look ahead for just the `(` part... that way your rule doesn't eat the `(` and it can be matched by ARGUMENTS later  or VALUE_CONTAINER...
- We need to add a rule BEFORE this one to match the BUILT_IN_GLOBALS functions and still highlight them as keywords properly.
- Then this rule would come after, prevent highlighting any other keywords as function names.
- And then technically this same fix would need to be applied to Javascript also (or at the very least an issue opened).

That's off the top of my head.

Reply:
You may also need to run the tests and see why they are currently red.  I'm guessing you either actually fixed something that was actually incorrectly highlighted in a test case or else you've accidentally broken some of the tests with these rule additions.

Reply:
Hang in there, try not to get discouraged.  Often contributing a small fix isn't nearly as simple as you first though... particularly once you get into the grammar rules, and inclusion, exclusion, etc... there are often unintended side effects, etc... so it takes time to work thru all those, cover all the cases, etc.

That's why we have the `markup` tests, to help prevent unintended regressions when changes like this are made, etc... so if you aren't running the tests you probably really should be.

Reply:
I'm not too familiar with `highlight.js`, so it's sometimes trial and error.
I use Windows OS, it fails for me and I don't have time to fix another test task.

Reply:
If you wanted to pull out this last part and just go with the first two fixes (assuming they pass tests on their own) then we could get those merged and add an issue to track the other stuff...

You'd still want to add a quick CHANGES.md mention.

Reply:
@yyyc514
I've come up to something like this
```js
begin: '\\b(?<!\\.)' + IDENT_RE + '(?=[\\(])', relevance: 0
```
but still ``(`` get's eaten somehow, so until next try to fix it, I don't know how to prevent this.

Reply:
That looks right regarding look ahead. Look behind is not well supported yet so we cannot use that.

Reply:
Could you suggest in what direction I should push?

Reply:
I don't know what can be done if positive lookahead doesn't work

Reply:
> I don't know what can be done if positive lookahead doesn't work

It most definitely works in general, we use it in many places.  What might help here is if you wrote a failing markup test and then pushed your current code AND the failing markup tests and then I could take a look at it locally.

And also please be explicit in why you say it's "not working".

Reply:
`test/markup/typescript/identifiers_that_include_keywords.txt` might be a good place to start.

Reply:
> I use Windows OS, it fails for me and I don't have time to fix another test task.

If you can't run the tests though I'm really not sure what to tell you - this is going to be a major pain to check all the results by hand for all cases.  Is there a particular reason you can run the test suite?  It should be possible on Windows if you have Node.js, etc installed.

Reply:
I fixed it in different way and tests have passed, but take a look again if it's ok in current shape.
After that I'll focus on making other necessary preparations.